src/audio/tensorflow/speech.cc

```diff
@@ -0,0 +1,153 @@
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// Copyright(c) 2025 Intel Corporation. All rights reserved.
+
+#include <algorithm>
+#include <cstdint>
+#include <iterator>
+
+#include "tensorflow/lite/core/c/common.h"
+#include "tensorflow/lite/micro/micro_interpreter.h"
+#include "tensorflow/lite/micro/micro_log.h"
+#include "tensorflow/lite/micro/micro_mutable_op_resolver.h"
+#include "tensorflow/lite/micro/testing/micro_test.h"
+#include "speech.h"
+
+// hard code the model today
+#include "micro_speech_quantized_model_data.h"
+
+// The following values are derived from values used during model training.
+// If you change the way you preprocess the input, update all these constants.
+//constexpr int kAudioSampleFrequency = TFLM_SAMPLE_RATE;
+static constexpr int kFeatureSize = TFLM_FEATURE_SIZE;
+static constexpr int kFeatureCount = TFLM_FEATURE_COUNT;
+static constexpr int kFeatureElementCount = TFLM_FEATURE_ELEM_COUNT;
+
+// Arena size is a guesstimate, followed by use of
+// MicroInterpreter::arena_used_bytes() on both the AudioPreprocessor and
+// MicroSpeech models and using the larger of the two results.
+static constexpr size_t kArenaSize = 28584;  // xtensa p6
+alignas(16) static uint8_t g_arena[kArenaSize];
+
+// type for features
+using Features = int8_t[kFeatureCount][kFeatureSize];
+
+// inference
+static const tflite::Model *model;
+static TfLiteTensor *input;
+static TfLiteTensor *output;
+static tflite::MicroInterpreter *interpreter;
+
+using MicroSpeechOpResolver = tflite::MicroMutableOpResolver<4>;
+static MicroSpeechOpResolver *op_resolver;
+
+// Adding more kernels is quite efficient. TODO add more
+int RegisterOps(MicroSpeechOpResolver *op_resolver) {
+	TF_LITE_ENSURE_STATUS(op_resolver->AddReshape());
+	TF_LITE_ENSURE_STATUS(op_resolver->AddFullyConnected());
+	TF_LITE_ENSURE_STATUS(op_resolver->AddDepthwiseConv2D());
+	TF_LITE_ENSURE_STATUS(op_resolver->AddSoftmax());
+	return 0;
+}
+
+int TF_InitOps(struct tf_classify *tfc)
+{
+	op_resolver = new MicroSpeechOpResolver();
+	if (RegisterOps(op_resolver) != 0) {
+		tfc->error = "register ops failed";
+		return -EINVAL;
+	}
+
+	// create the interpreter
+	interpreter = new tflite::MicroInterpreter(model, *op_resolver,
+						   g_arena, kArenaSize);
+
+	// and allocate the tensors
+	if (interpreter->AllocateTensors() != kTfLiteOk) {
```

Comment:
This is more of a tflite API question, but wondering can "interpreter" be NULL/nullptr here? Maybe this is ok and the new operator is just overloaded.